.github/workflows/setup_environment.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+
+# Install packages
+sudo snap install core
+sudo apt-get remove -qy lxd lxd-client
+sudo snap install lxd --channel=latest/stable
+sudo snap refresh lxd --channel=latest/stable
+
+# LXD setup
+lxd waitready
+lxd init --auto
+chmod a+wr /var/snap/lxd/common/lxd/unix.socket
+lxc network set lxdbr0 ipv6.address none
+sudo usermod -a -G lxd ubuntu
+newgrp lxd
+
+# Python setup
+sudo apt-get update -yqq
+sudo apt-get install -yqq python3-pip
```

Comment:
To be consistent for other charms, do we need to explicitly call out python 3.8 here?  Or does this guarantee a python 3.8 install?  If this will always give python 3.8, maybe still comment on that here since we're particular about the version due to pip-compile+pylibjuju?